src/test/java/uk/nhs/hee/trainee/details/migration/DeleteDuplicateTraineeProfilePlacementsTest.java

```diff
@@ -0,0 +1,79 @@
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.trainee.details.migration;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.util.AssertionErrors.assertTrue;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.result.UpdateResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+
+class DeleteDuplicateTraineeProfilePlacementsTest {
+  private static final String PROFILE_COLLECTION = "TraineeProfile";
+  private DeleteDuplicateTraineeProfilePlacements migration;
+  private MongoTemplate template;
+
+  @BeforeEach
+  void setUp() {
+    template = mock(MongoTemplate.class);
+    migration = new DeleteDuplicateTraineeProfilePlacements(template);
+  }
+
+  @Test
+  void shouldDeleteDuplicateTraineeProfilePlacements() {
+    final Update update =
+        new Update().pull("placements",
+            new BasicDBObject("tisId", new BasicDBObject("$regex", "^\\w{24}$")));
+    ArgumentCaptor<Query> queryCaptor = ArgumentCaptor.forClass(Query.class);
+    ArgumentCaptor<Update> updateCaptor = ArgumentCaptor.forClass(Update.class);
+    UpdateResult result = mock(UpdateResult.class);
+
+    when(template.updateMulti(
+        queryCaptor.capture(),
+        updateCaptor.capture(),
+        eq(PROFILE_COLLECTION)))
+        .thenReturn(result);
+    when(result.getModifiedCount()).thenReturn(1L);
+
+    migration.migrate();
+
+    Query queryUsed = queryCaptor.getValue();
+    assertTrue("Unexpected query.", queryUsed.getQueryObject().isEmpty());
+    Update updateUsed = updateCaptor.getValue();
+    assertTrue("Unexpected placement deletion", updateUsed.equals(update));
```

Comment:
I'd rather we used Hamcrest matchers here as they usually generate better failure messages.

Reply:
Thanks, it has been updated to use Hamcrest matchers.